frontend/src/metabase/api/action.ts

```diff
@@ -1,21 +1,73 @@
+import _ from "underscore";
+
 import type {
+  CreateActionRequest,
   GetActionRequest,
+  ListActionsRequest,
+  UpdateActionRequest,
   WritebackAction,
   WritebackActionId,
 } from "metabase-types/api";
 import type { GetPublicAction } from "metabase-types/api/actions";
 
 import { Api } from "./api";
-import { idTag, invalidateTags, listTag } from "./tags";
+import {
+  idTag,
+  invalidateTags,
+  listTag,
+  provideActionListTags,
+  provideActionTags,
+} from "./tags";
 
 export const actionApi = Api.injectEndpoints({
   endpoints: builder => ({
+    listActions: builder.query<WritebackAction[], ListActionsRequest>({
+      query: params => ({
+        method: "GET",
+        url: `/api/action`,
+        params,
+      }),
+      providesTags: (collections = []) => provideActionListTags(collections),
+    }),
     getAction: builder.query<WritebackAction, GetActionRequest>({
       query: ({ id }) => ({
         method: "GET",
         url: `/api/action/${id}`,
       }),
-      providesTags: action => (action ? [idTag("action", action.id)] : []),
+      providesTags: action => (action ? provideActionTags(action) : []),
+    }),
+    createAction: builder.mutation<WritebackAction, CreateActionRequest>({
+      query: body => ({
+        method: "POST",
+        url: "/api/action",
+        body,
+      }),
+      invalidatesTags: (action, error) =>
+        action ? [...invalidateTags(error, [listTag("action")])] : [],
```

Comment:
Can we just write?

```suggestion
        action ? invalidateTags(error, [listTag("action")]) : [],
```

Reply:
6568d593090107429e3196443f19b935c3da9c23